go/libraries/doltcore/rowconv/super_schema_test.go

```diff
@@ -0,0 +1,259 @@
+				"0_UUID":   TagKindPair{0, types.UUIDKind},
+				"1_String": TagKindPair{1, types.StringKind},
+				"2_String": TagKindPair{2, types.StringKind},
+				"3_String": TagKindPair{3, types.StringKind},
+				"3_Int":    TagKindPair{schema.ReservedTagMin, types.IntKind},
+			}),
+		},
+		{
+			"tag type conflict",
+			[]schema.Schema{
+				mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr),
+				mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, addrColTag3TypeStr),
+				mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, titleColTag3TypeStr),
+			},
+			mustSchemaFromTagAndKind(map[string]TagKindPair{
+				"0_UUID":   TagKindPair{0, types.UUIDKind},
+				"1_String": TagKindPair{1, types.StringKind},
+				"2_String": TagKindPair{2, types.StringKind},
+				"3_String": TagKindPair{3, types.StringKind},
+			}),
+		},
+		{
+			"multiple tag conflicts",
+			[]schema.Schema{
+				mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr),
+				mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, addrColTag3TypeStr),
+				mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, ageColTag3TypeInt),
+				mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, addrColTag3TypeStr, ageColTag4TypeInt),
+				mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, addrColTag3TypeStr, ageColTag4TypeUint),
+			},
+			mustSchemaFromTagAndKind(map[string]TagKindPair{
+				"0_UUID":   TagKindPair{0, types.UUIDKind},
+				"1_String": TagKindPair{1, types.StringKind},
+				"2_String": TagKindPair{2, types.StringKind},
+				"3_String": TagKindPair{3, types.StringKind},
+				"3_Int":    TagKindPair{schema.ReservedTagMin, types.IntKind},
+				"4_Int":    TagKindPair{4, types.IntKind},
+				"4_Uint":   TagKindPair{schema.ReservedTagMin + 1, types.UintKind},
+			}),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ssg := NewSuperSchemaGen()
+
+			for _, sch := range test.schemas {
+				err := ssg.AddSchema(sch)
+				require.NoError(t, err)
+			}
+
+			result, err := ssg.GenerateSuperSchema()
+			require.NoError(t, err)
+
+			eq, err := schema.SchemasAreEqual(result, test.expected)
+			require.NoError(t, err)
+			assert.True(t, eq)
+		})
+	}
+}
+
+func TestSuperSchemaFromHistory(t *testing.T) {
+	const tblName = "test_table"
+	initialSch := mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr)
+	addAddrAt3Sch := mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, addrColTag3TypeStr)
+	addAgeAt3Sch := mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, ageColTag3TypeInt)
+	readdAgeAt4Sch := mustSchema(idColTag0TypeUUID, firstColTag1TypeStr, lastColTag2TypeStr, addrColTag3TypeStr, ageColTag4TypeUint)
+
+	ctx := context.Background()
+	dEnv := envtestutils.CreateInitializedTestEnv(t, ctx)
+
+	history := envtestutils.HistoryNode{
```

Comment:
Probably want to have a test case that involves a merge, right?

Reply:
not supported yet.  Maybe one day.